dfastmi/io/RiverParameterData.py

```diff
@@ -219,183 +226,175 @@ def collect_values1(
-                    except:
-                        vals = ()
-                    if len(vals) != 1:
-                        raise Exception(
-                            'Reading {} for reach {} on {} returns "{}". Expecting {} values.'.format(
-                                key, i+1, self._branches[ib], val, 1
-                            )
-                        )
-                    fval = vals[0]
-                values_per_branch.append(fval)
+        Arguments
+        ---------
+        key : str
+            The name of the parameter for which the values are to be retrieved.
+        branch : Branch
+            The branch where we want to read the parameter from
 
-            all_values.append(values_per_branch)
 
-        return all_values
+        Raises
+        ------
+        Exception
+            If the number of values read from the file doesn't match 1.
+
+        Returns
+        -------
+        data : List[List[float]]
+            A list of lists. Each list contains per reach within the corresponding
+            branch one float.
+        """
+        self.is_initialized()
+        val = self.__read_value(key, branch, reach)
+        if val == "" and default is not None:
+            fval = default
+        else:
+            try:
+                vals = tuple(float(x) for x in val.split())
+            except:
+                vals = ()
+            if len(vals) != 1:
+                raise Exception(
+                    'Reading {} for reach {} on branch {} returns "{}". Expecting {} values.'.format(
+                        key, reach.name, branch.name, val, 1
+                    )
+                )
+            fval = vals[0]
+        return fval     
     
-    def collect_values2(
-        self,
-        key: str,
-        default: Optional[Tuple[float, float]] = None,
-    ) -> List[List[Tuple[float, float]]]:
+    def read_key_tuple_float_float(
+            self,
+            key: str,
+            branch: Branch,
+            reach: Reach,
+            default: Optional[Tuple[float, float]] = None
+        ) -> Tuple[float, float]:
         """
-        Collect river parameter data from river configuration object.
-
-        This routines collects entries of type Tuple[float, float].
+        This routines collects entries of type float.
 
         Arguments
         ---------
         key : str
             The name of the parameter for which the values are to be retrieved.
-        default : Optional[Tuple[float, float]]
-            Default tuple if not specified in file.
+        branch : Branch
+            The branch where we want to read the parameter from
+
 
         Raises
         ------
         Exception
-            If the number of values read from the file doesn't match 2.
+            If the number of values read from the file doesn't match 1.
 
         Returns
         -------
-        data : List[List[Tuple[float, float]]]
+        data : List[List[float]]
```

Comment:
repeat sentence above

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112